packages/webdriverio/src/commands/browser/switchWindow.ts

```diff
@@ -69,6 +69,13 @@ export async function switchWindow (
         return !!target.match(matcher)
     }
 
+    // In WebDriver Classic, after closing a window, we need to switch to a valid window first
+    if (!this.isBidi && (!currentWindow || !tabs.includes(currentWindow))) {
+        // Switch to the first available window before proceeding
+        await this.switchToWindow(tabs[0])
+        contextManager.setCurrentContext(tabs[0])
```

Comment:
@christian-bromann correct me if I am wrong but the context manager is only used in BiDi right?